File: gwi_ti_test.php
<?php
/**
 * gwi_ti_test.php
 * Test harness for for interacting with the GhostWolf Threat Intelligence APIs
 * Not fully parametereized, illustrative purposes only
 *
 * PHP version 5.5+
 *
 * @author		jfitterer@GhostWolfindustries.com
 * @copyright	2014-2019 GhostWolf Industries, Inc.
 * @license		
 * @version		0.1
 * @since		  v0.1
 */

/*
 * Build the request to obtain an access token
 */
$pVars = array(
	'client_id' => 'user@domain.com',
	'client_secret' => 'API_KEY',
	'grant_type' => 'client_credentials'
);

$ch = curl_init();
curl_setopt($ch, CURLOPT_URL,"https://ti-api.ghostwolfindustries.com/token");
curl_setopt($ch, CURLOPT_POST, 1);
curl_setopt($ch, CURLOPT_POSTFIELDS,$pVars);  //Post Fields
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

/*
 * With the raw response in hand, extract the access_token and close the connection
 */
$tResponse = curl_exec ($ch);
$tResponse = json_decode($tResponse);
$token = $tResponse->access_token;
curl_close ($ch);

/*
 * Using the access token, issue a GET request to determine if the subject ip 128.75.222.116
 * is in the GhostWolf Threat ntel database
 * 
 */
$header = array(
		'Authorization: Bearer '.$token
);
$ch = curl_init();
curl_setopt($ch, CURLOPT_HTTPHEADER, $header);
curl_setopt($ch, CURLOPT_URL,"https://ti-api.ghostwolfindustries.com/ip/128.75.222.116");
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
$aResponse = curl_exec ($ch);

/*
 * Reformat and print the output
 */
$aResponse = json_decode($aResponse);
print_r($aResponse);
curl_close ($ch);

?>

File: gwi_ti_test.py
'''
  gwi_ti_test.php
  Test harness for for interacting with the Hoplite Threat Intelligence APIs
  Not fully parametereized, illustrative purposes only
 
  @author	jfitterer@ghostwolfindustries.com
  @copyright	2014-2019 GhostWolf Industries, Inc.
  @license		
  @version	0.1
  @since	v0.1
'''
import requests
import json
import pprint

'''
  Build the request to obtain an access token
'''
data = {"client_id": "user@domain.com", "client_secret": "XXXX-XXXX-XXXX-XXXXX-XXXXX", "grant_type": "client_credentials"}

'''
  With the raw response in hand, extract the access_token and close the connection
'''
tResponse = requests.post('https://ti-api.ghostwolfindustries.com/token', data)
json_data = json.loads(tResponse.text)
token = json_data['access_token']

'''
  Using the access token, issue a GET request to determine if the subject ip 128.75.222.116 is in the Hoplite Threat Intel database
'''
aResponse = requests.get('https://ti-api.ghostwolfindustries.com/ip/128.75.222.116', headers={'Authorization': 'Bearer ' + token})

'''
  Reformat and print the output
'''
json_data = json.loads(aResponse.text)
pp = pprint.PrettyPrinter(indent=4)
pp.pprint(json_data)
